i-VRESSE/bartended-haddock3
# Remix auth

Fixes https://github.com/i-VRESSE/bartender/issues/54

Should be merged together with https://github.com/i-VRESSE/bartender/pull/73

TODO

- [x] store private key for making bartender tokens
- [x] use remix auth for local users with remix-auth-form
- [x] use remix-auth-oauth2 for
  - [x] orcid sandbox
  - [x] orcid prod
  - [x] egi-checkin
- [x] associate local user with oauth account aka mail matching
- [x] regenerated app/bartender-client against https://github.com/i-VRESSE/bartender/pull/73
- [x] dont generate bartender token every time, but store/fetch from db
- [x] enforce password length with zod
- [x] store users in postgresql instead of sqlite
- [x] Replace expertise level seed with enum as new level needs code change, so why not have levels as code as well.
- [x] document oauth2 provider setup
- [x] docker-compose with bartender and haddock3-webapp dbs + rsa key pair mount
- [x] explain remix + prisma choice 
    - [x] workflow builder works, while in webpack based meta framework the build chokes
    - [x] follow remix blue stack template, for least dev surprises
- [x] Upgraded to latest remix
    - [x] Use v2 route file naming
- [x] use prisma migrations to create tables in db
- [x] Calculate gravatar hash

To test:
1. Start bartender web service from https://github.com/i-VRESSE/bartender/tree/asym-jwt-only branch with public key.
2. Setup webapp according to development instructions in README.md
3. Register + login + give yourself an expertise level
4. Check on build page that topoaa node has options for your selected expertise level
5. Configure social login GitHub, Orcid or EGI and try to login with those accounts. PS. Make sure you a logged out from the webapp before trying to log in again.
6. 
7. Start webapp with docker compose using  Docker instructions in README.m
